inspirehep/alembic/2f5368ff6d20_add_index_on_legacy_records_mirror_table.py

```diff
@@ -0,0 +1,36 @@
+#
+# This file is part of Invenio.
+# Copyright (C) 2016-2018 CERN.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Add index on legacy_records_mirror"""
+
+from __future__ import absolute_import, division, print_function
+
+from alembic import op
+
+
+# revision identifiers, used by Alembic.
+revision = '2f5368ff6d20'
+down_revision = 'eaab22c59b89'
+branch_labels = ()
+depends_on = None
+
+
+def upgrade():
+    """Upgrade database."""
+    op.create_index(
+        'ix_legacy_records_mirror_collection_valid',
+        'legacy_records_mirror',
+        ['collection', 'valid'],
```

Comment:
it's probably more useful to have the index in the opposite direction, that is on `('valid', 'collection')`, so that it can be used instead of the current index on `valid` (that you can remove, as it would do double duty with this one).

You can think about an index as something like a phone book (if you know what a phonebook is). That is an index on `(last name, first name)`, and can be used to obtain quickly the phone number of a person if you know either:
* the last name
* the last name and the first name
but it's of no use if you know only the first name.

Here, we want something that can be used to get quickly invalid records, or invalid records and their collection, but we don't need the collection of a record without its validity. That's why the other order is better.

Reply:
okay! Thanks!